frontend/components/LiveQuery/SelectTargets.tsx

```diff
@@ -253,6 +279,50 @@ const SelectTargets = ({
     }
   );
 
+  // Ensure that the team or label list is expanded on the first load only if a hidden entity is already selected
+  const shouldExpandList = (
+    list: ISelectLabel[] | ISelectTeam[],
+    truncatedList: ISelectLabel[] | ISelectTeam[]
+  ) => {
+    return list.some(
+      (entity) =>
+        !truncatedList.some(
+          (truncatedEntity) => truncatedEntity.id === entity.id
+        )
+    );
+  };
+
+  const expandListsOnInitialLoad = () => {
+    if (!isMountedRef.current && teams && labels) {
+      const truncatedLabels =
+        labels?.other?.slice(
+          0,
+          getTruncatedLength(labels?.other, SECTION_CHARACTER_LIMIT)
+        ) || [];
+      const truncatedTeams =
+        teams?.slice(0, getTruncatedLength(teams, SECTION_CHARACTER_LIMIT)) ||
+        [];
+
+      if (shouldExpandList(targetedLabels, truncatedLabels)) {
+        setIsLabelsListExpanded(true);
+      }
+
+      if (shouldExpandList(targetedTeams, truncatedTeams)) {
+        setIsTeamListExpanded(true);
+      }
+
+      isMountedRef.current = true;
+    }
+  };
+
+  useEffect(expandListsOnInitialLoad, [
+    targetedTeams,
+    targetedLabels,
+    labels,
+    teams,
+    isMountedRef,
+  ]);
```

Comment:
Since `targetedTeams` and `targetedLabels` are held as parent states and `labels` and `teams` come
from an API call and won't be changed on this page, is it necessary to use `useEffect` here or could
we just have all the logic in `expandListsOnInitialLoad` sit outside a callback/useEffect?

Reply:
Ok for this one it was tricky because it only should run after it's mounted when teams and labels is loaded. The solution to use `isMountedRef` ensures that the `expandListsOnInitialLoad` function is called only once, when the component is first mounted and the necessary teas/labels is available.